File: SortVision/public/code/selection/java/selectionSort.java
public class SelectionSort {
    // Time: Best/Avg/Worst O(n^2), Space: O(1)
    public static void selectionSort(int[] arr) {
        if (arr == null || arr.length <= 1) return;

        for (int i = 0; i < arr.length - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[j] < arr[minIdx]) minIdx = j;
            }
            if (minIdx != i) {
                int t = arr[i];
                arr[i] = arr[minIdx];
                arr[minIdx] = t;
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {64, 25, 12, 22, 11};
        selectionSort(arr);
        for (int x : arr) System.out.print(x + " ");
        System.out.println();
    }
}

File: SortVision/public/code/selection/scala/selectionSort.scala
object SelectionSort {
  // Time: Best/Avg/Worst O(n^2), Space: O(1)
  def selectionSort(arr: Array[Int]): Array[Int] = {
    for (i <- 0 until arr.length - 1) {
      var minIdx = i
      for (j <- i + 1 until arr.length) {
        if (arr(j) < arr(minIdx)) minIdx = j
      }
      if (minIdx != i) {
        val t = arr(i)
        arr(i) = arr(minIdx)
        arr(minIdx) = t
      }
    }
    arr
  }

  def main(args: Array[String]): Unit = {
    val arr = Array(64, 25, 12, 22, 11)
    println(selectionSort(arr).mkString(", "))
  }
}
